src/js/components/Calendar/Calendar.js

```diff
@@ -448,14 +484,28 @@ const Calendar = forwardRef(
       },
       [onReference, bounds],
     );
+
     const selectDate = useCallback(
       (selectedDate) => {
+        const formatDate = (d) => {
```

Comment:
Also, can we rename this to be a bit more descriptive about what we are formatting it to. For instance "formatToLocalWithTimestamp"?